### PR TITLE
[Improvement][Alert] Alert plugin enhance fail message

### LINKED ISSUE #15023: [Improvement][Alert] Alert plugin enhance fail message
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Alert plugin enhance fail message, user can see the error reason in db

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qingwli**
- close #15023

### CODE REVIEW

**Comment by qingwli** [comment]
Kudos, SonarCloud Quality Gate passed!    Quality Gate passed

Bug A [0 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=15024&resolved=false&types=BUG)
Vulnerability A [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=15024&resolved=false&types=VULNERABILITY)
Security Hotspot A [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=15024&resolved=false&types=SECURITY_HOTSPOT)
Code Smell A [0 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=15024&resolved=false&types=CODE_SMELL)

92.3% [92.3% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=15024&metric=new_coverage&view=list)
0.0% [0.0% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=15024&metric=new_duplicated_lines_density&view=list)

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public AlertResult send(String msg) {
         } catch (Exception e) {
             log.error("send http alert msg  exception : {}", e.getMessage());
             alertResult.setStatus("false");
-            alertResult.setMessage("send http request  alert fail.");
+            alertResult.setMessage("send http request  alert fail." + e.getMessage());
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java]:
```suggestion
            alertResult.setMessage("send http request alert fail:" + e.getMessage());
```

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-email/src/main/java/org/apache/dolphinscheduler/plugin/alert/email/MailSender.java [hunk 2]
```diff
@@ -405,7 +405,7 @@ public void deleteFile(File file) {
      */
     private void handleException(AlertResult alertResult, Exception e) {
         log.error("Send email to {} failed", receivers, e);
-        alertResult.setMessage("Send email to {" + String.join(",", receivers) + "} failed，" + e.toString());
+        alertResult.setMessage("Send email to {" + String.join(",", receivers) + "} failed，" + e.getMessage());
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-email/src/main/java/org/apache/dolphinscheduler/plugin/alert/email/MailSender.java]:
```suggestion
        alertResult.setMessage("Send email to {" + String.join(",", receivers) + "} failed:" + e.getMessage());
```

Comment (Radeity) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-email/src/main/java/org/apache/dolphinscheduler/plugin/alert/email/MailSender.java]: You can keep consistency and use `String.format`.

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-email/src/main/java/org/apache/dolphinscheduler/plugin/alert/email/MailSender.java]: Done, Please check again

**Review by Radeity — APPROVED** [review]
LGTM

**Pull request merged**